processor/routingprocessor/factory_test.go

```diff
@@ -136,6 +143,55 @@ func TestShouldNotFailWhenNextIsProcessor(t *testing.T) {
 	assert.NotNil(t, exp)
 }
 
+func TestProcessorDoesNotFailToBuildExportersWithMultiplePipelines(t *testing.T) {
+	// prepare
+	factories, err := componenttest.NopFactories()
+	assert.NoError(t, err)
+
+	processorFactory := NewFactory()
+	factories.Processors[typeStr] = processorFactory
+
+	otlpExporterFactory := otlpexporter.NewFactory()
+	factories.Exporters["otlp"] = otlpExporterFactory
+
+	otlpConfig := &otlpexporter.Config{
+		ExporterSettings: config.NewExporterSettings(config.NewComponentID("otlp")),
+		GRPCClientSettings: configgrpc.GRPCClientSettings{
+			Endpoint: "example.com:1234",
+		},
+	}
+
+	otlpTracesExporter, err := otlpExporterFactory.CreateTracesExporter(context.Background(), componenttest.NewNopExporterCreateSettings(), otlpConfig)
+	require.NoError(t, err)
+
+	otlpMetricsExporter, err := otlpExporterFactory.CreateMetricsExporter(context.Background(), componenttest.NewNopExporterCreateSettings(), otlpConfig)
+	require.NoError(t, err)
+
+	host := &mockHost{
+		Host: componenttest.NewNopHost(),
+		GetExportersFunc: func() map[config.DataType]map[config.ComponentID]component.Exporter {
+			return map[config.DataType]map[config.ComponentID]component.Exporter{
+				config.TracesDataType: {
+					config.NewComponentID("otlp/traces"):   otlpTracesExporter,
+				},
+				config.MetricsDataType: {
+					config.NewComponentID("otlp/metrics"):   otlpMetricsExporter,
+				},
+			}
+		},
+	}
+
+	cfg, err := servicetest.LoadConfigAndValidate(filepath.Join("testdata", "config_multipipelines.yaml"), factories)
+	assert.NoError(t, err)
+
+	for _, cfg := range cfg.Processors {
+		exp := newProcessor(zap.NewNop(), cfg)
+		err = exp.Start(context.Background(), host)
+		// assert that no error is thrown due to multiple pipelines and exporters not using the routing processor
+		assert.NoError(t, err)
```

Comment:
I think it practically doesn't change much but for convenience perhaps it would be good to also do the shutdown, e.g. something along
```suggestion
		assert.NoError(t, err)
		assert.NoError(t, exp.Shutdown(context.Background())
```

Reply:
Makes sense, just added the shutdown